app/views/play/level/ScreenReaderSurfaceView.coffee

```diff
@@ -0,0 +1,116 @@
+require('app/styles/play/level/screen-reader-surface-view.sass')
+CocoView = require 'views/core/CocoView'
+template = require 'app/templates/play/level/screen-reader-surface'
+utils = require 'core/utils'
+
+module.exports = class ScreenReaderSurfaceView extends CocoView
+  id: 'screen-reader-surface-view'
+  template: template
+
+  subscriptions:
+    'surface:update-screen-reader-map': 'onUpdateScreenReaderMap'
+    'camera:zoom-updated': 'onUpdateScreenReaderMap'
+
+  constructor: (options) ->
+    super options
+
+  afterInsert: ->
+    super()
+    $(window).on('keydown', @onKeyEvent)
+    $(window).on('keyup', @onKeyEvent)
+    @updateScale()
+
+  destroy: ->
+    $(window).off('keydown', @onKeyEvent)
+    $(window).off('keyup', @onKeyEvent)
+    super()
+
+  onUpdateScreenReaderMap: (e) ->
+    if e?.grid
+      @grid = e.grid
+      @gridChars = @grid.toSimpleMovementChars(true, false)
+      @gridNames = @grid.toSimpleMovementNames()
+      #console.log @grid, @gridNames
+      #console.log @grid.toString(true, false)
+    @updateCells()
+    @updateScale()
+
+  onKeyEvent: (e) =>
+    event = _.pick(e, 'type', 'key', 'keyCode', 'ctrlKey', 'metaKey', 'shiftKey')
+    #console.log 'got event', event, @highlight, /^arrow/.test(event.key.toLowerCase()), event.key.toLowerCase()
+    # TODO: only do this if we have the map area focused
+    return unless @highlight
+    return unless /^arrow/.test event.key.toLowerCase()
+    return unless event.type is 'keydown'
+    newHighlight = {row: @highlight.row, col: @highlight.col}
+    switch event.key.toLowerCase()
+      when 'arrowleft'  then --newHighlight.col
+      when 'arrowright' then ++newHighlight.col
+      when 'arrowup'    then --newHighlight.row
+      when 'arrowdown'  then ++newHighlight.row
+    pan = Math.max -1, Math.min 1, -1 + 2 * newHighlight.col / (@gridNames[0].length - 1)
+    if (newHighlight.$cell = @mapCells[newHighlight.row]?[newHighlight.col]) and newHighlight.$cell.previousChar.trim()
+      # There's something here (a movable area and/or an object)
+      @highlight.$cell.removeClass 'highlighted'
+      newHighlight.$cell.addClass 'highlighted'
+      @highlight = newHighlight
+      update = @gridNames[@highlight.row][@highlight.col]
+      update = update.replace /,? ?Dot,? ?/g, ''
+      @$el.find('.map-screen-reader-live-updates').text(update)
+      @playSound 'game-menu-tab-switch', 1, 0, null, pan  # temporary sfx we already have
+    else
+      # There's nothing here, and the hero can't move here--don't move the highlight cursor
+      @playSound 'menu-button-click', 1, 0, null, pan  # temporary sfx we already have, need something more different from success sound
+
+  updateCells: ->
```

Comment:
this function seems like an algorithm code 😅  Can you add a description to what this function actually does or better naming and split it into small parts?